src/ui/components/filters/filterboxcomponent.js

```diff
@@ -40,13 +40,13 @@ class FilterBoxConfig {
      * If true, show a "reset all" button to reset all facets
      * @type {boolean}
      */
-    this.resetFilters = config.resetFacets === undefined ? true : config.resetFacets;
+    this.resetFilters = config.resetFilters === undefined ? !config.searchOnChange : config.resetFilters;
```

Comment:
This change does not break facets, before facets were passing a resetFacets param both as resetFilters AND as resetFacets, see https://github.com/yext/answers/blob/master/src/ui/components/filters/facetscomponent.js#L209

Reply:
resetFilter is also listening to resetFacet. Can we correct that too?

Reply:
Where is the default of !config.searchOnChange coming from? Is that maintaining backwards compatibility? It looks like the default was true before.

Reply:
The default WAS true before. But in the template, the reset button would only render if the showApplyButton is true, and showApply is set to !config.searchOnChange

Reply:
Separate PR is fine! Thanks for taking care of it.